store/tikv/tikvrpc/tikvrpc.go

```diff
@@ -746,22 +746,36 @@ func (resp *CopStreamResponse) Recv() (*coprocessor.Response, error) {
 // Close closes the CopStreamResponse object.
 func (resp *CopStreamResponse) Close() {
 	atomic.StoreInt64(&resp.Lease.deadline, 1)
+	// We also call cancel here because CheckStreamTimeoutLoop
+	// is not guaranteed to cancel all items when it exits.
+	if resp.Lease.Cancel != nil {
+		resp.Lease.Cancel()
+	}
 }
 
 // CheckStreamTimeoutLoop runs periodically to check is there any stream request timeouted.
 // Lease is an object to track stream requests, call this function with "go CheckStreamTimeoutLoop()"
-func CheckStreamTimeoutLoop(ch <-chan *Lease) {
+// It is not guaranteed to call every Lease.Cancel() putting into channel when exits.
+// If grpc-go supports SetDeadline(https://github.com/grpc/grpc-go/issues/2917), we can stop using this method.
+func CheckStreamTimeoutLoop(ch <-chan *Lease, done <-chan struct{}) {
 	ticker := time.NewTicker(200 * time.Millisecond)
 	defer ticker.Stop()
 	array := make([]*Lease, 0, 1024)
 
 	for {
 		select {
-		case item, ok := <-ch:
-			if !ok {
-				// This channel close means goroutine should return.
-				return
+		case <-done:
+		drainLoop:
+			// Try my best cleaning the channel to make SendRequest which is blocking by it continues.
+			for {
+				select {
+				case <-ch:
+				default:
+					break drainLoop
```

Comment:
I think the outside `for` loop is enough?

Reply:
Consider a huge number of `SendRequest` is waiting for channel `ch`, it may be the last chance to help them finish.